tests/functions/test_conditional_join.py

```diff
@@ -305,7 +305,7 @@ def test_single_condition_less_than_ints_extension_array(df, right):
     left_on, right_on = ["A", "Integers"]
     expected = (
         df.merge(right, how="cross")
-        .query(f"{left_on} < {right_on}")
+        .query(f"{left_on} < {right_on}", engine="python")
```

Comment:
Curious, @samukweku, according to the `pandas` [docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html):

> You can change the semantics of the expression by passing the keyword argument parser='python'. This enforces the same semantics as evaluation in Python space. Likewise, you can pass engine='python' to evaluate an expression using Python itself as a backend. This is not recommended as it is inefficient compared to using numexpr as the engine.

Would it be more advantageous if we left the `engine` kwarg to be the default `numexpr`?

Reply:
numexpr for some reason was failing, hence the `engine=python` fix. Made the changes, using pandas' loc instead. I'll make the changes for the others in another PR.